applications/common/frontend/src/components/Image/image.ts

```diff
@@ -16,16 +16,65 @@
 
 import Viewer from "viewerjs";
 
-const imagesWithLightbox = document.querySelectorAll<HTMLImageElement>(
-  ".image--lightbox"
-);
-imagesWithLightbox.forEach((image) => {
-  new Viewer(image, {
+// We may extend this array later with additional pages that need to have an image preview carousel
+const imageCarouselTemplates = ["template-blogarticlepage"];
+
+const createSeparatePreviews = (imageList) => {
+  imageList.forEach((image) => {
+    new Viewer(image, {
+      navbar: false,
+      toolbar: false,
+      zoomable: false,
+      movable: false,
+      title: false,
+      url: 'data-lightbox-asset',
+      show() {
+        document.querySelector("html").style.overflowY = "hidden";
+      },
+      hidden() {
+        document.querySelector("html").style.overflowY = "scroll";
+      }
+    });
+  });
+};
+
+const createPreviewCarousel = (imageContainer) => {
+  new Viewer(imageContainer, {
     navbar: false,
-    toolbar: false,
     zoomable: false,
     movable: false,
     title: false,
-    url: 'data-lightbox-asset'
+    scalable: false,
+    url: 'data-lightbox-asset',
+    toolbar: {
+      prev: 1,
+      next: 1,
+    },
+    show() {
+      document.querySelector("html").style.overflowY = "hidden";
+    },
+    hidden() {
+      document.querySelector("html").style.overflowY = "scroll";
+    }
+  });
+}
+
+if (imageCarouselTemplates.some(className => document.body.classList.contains(className))) {
+  const imageContainer = document.querySelector<HTMLElement>(".blogArticleContent");
```

Comment:
we can extend the `imageCarouselTemplates` list, but we can't change this class selector for the imageContainer. I think it won't fit all the templates in the future.

Reply:
Somehow I had to grab the one common parent of all images that should be in the carousel, because the ViewerJS library will only be able to pick them up this way. I added a `customClass` property to `applications/blogs/backend/src/main/resources/libs/kyanite/blogs/templates/blogarticlepage/initial/.content.json` and my idea was that we can add these classes to other templates (and run a Groovy script for already existing instances) that we want to extend this functionality to.
The other option was to create a not so pretty selector to get that specific column component.

Do you have any idea how could we do it differently?

Reply:
I think we could at last change the class name to something more general :) I mean, we could add some additional class like `imageContainer` and use it instead of `blogArticleContent`